### PR TITLE
Some improvement in Study and related widgets

### PR DESCRIPTION
**Pull request by samuelgarcia**

### CODE REVIEW

File: src/spikeinterface/widgets/comparison.py [hunk 1]
```diff
@@ -77,3 +75,87 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
...
+
+class AgreementMatrixWidget(BaseWidget):
+    """
+    Plots sorting comparison agreement matrix.
+
+    Parameters
+    ----------
+    sorting_comparison: GroundTruthComparison or SymmetricSortingComparison
+        The sorting comparison object.
+        Symetric or not.
```
Comment (zm711) [inline src/spikeinterface/widgets/comparison.py:90]:
```suggestion
        Can optionally be symmetric if given a SymmetricSortingComparison
```

File: src/spikeinterface/widgets/comparison.py [hunk 2]
```diff
@@ -77,3 +75,87 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
...
+    ordered: bool
+        Order units with best agreement scores.
+        This enable to see agreement on a diagonal.
```
Comment (zm711) [inline src/spikeinterface/widgets/comparison.py:93]:
```suggestion
        If True, agreement scores can be seen along a diagonal
```

File: src/spikeinterface/widgets/comparison.py [hunk 3]
```diff
@@ -77,3 +75,87 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
...
+class AgreementMatrixWidget(BaseWidget):
+    """
+    Plots sorting comparison agreement matrix.
+
+    Parameters
+    ----------
+    sorting_comparison: GroundTruthComparison or SymmetricSortingComparison
+        The sorting comparison object.
+        Symetric or not.
+    ordered: bool
```
Comment (zm711) [inline src/spikeinterface/widgets/comparison.py:91]:
```suggestion
    ordered: bool, default: True
```

File: src/spikeinterface/widgets/comparison.py [hunk 4]
```diff
@@ -77,3 +75,87 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
...
+        Order units with best agreement scores.
+        This enable to see agreement on a diagonal.
+    count_text: bool
```
Comment (zm711) [inline src/spikeinterface/widgets/comparison.py:94]:
```suggestion
    count_text: bool, default: True
```

File: src/spikeinterface/widgets/comparison.py [hunk 5]
```diff
@@ -77,3 +75,87 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
...
+        If True counts are displayed as text
+    unit_ticks: bool
```
Comment (zm711) [inline src/spikeinterface/widgets/comparison.py:96]:
```suggestion
    unit_ticks: bool, default: True
```

**Review by zm711 — COMMENTED** [review]
Just reviewing the docstring :)

**Comment by samuelgarcia** [comment]
@alejoe91 ready to merge on my side.

File: src/spikeinterface/widgets/gtstudy.py [hunk 6]
```diff
@@ -140,16 +134,18 @@ class StudyPerformances(BaseWidget):
     study: GroundTruthStudy
         A study object.
     mode: str
-        Which mode in "swarm"
+        Which mode in "ordered", "snr", "swarm"
```
Comment (zm711) [inline src/spikeinterface/widgets/gtstudy.py:137]: I don't know what 'swarm' is... Is this common knowledge or would it be beneficial to define this? How would a user know when to use 'ordered' vs 'snr' vs 'swarm'....

File: src/spikeinterface/widgets/gtstudy.py [hunk 7]
```diff
@@ -222,32 +253,88 @@ def __init__(
...
+
+        # ax0 = self.axes.flatten()[0]
+        # for ax in self.axes.flatten()[1:]:
+        #     ax.sharey(ax0)
+
+
+class StudySummary(BaseWidget):
+    """
+    Plot a summary of a ground truth study.
+    Internally do:
```
Comment (zm711) [inline src/spikeinterface/widgets/gtstudy.py:298]:
```suggestion
    Internally does:
```

File: src/spikeinterface/widgets/gtstudy.py [hunk 8]
```diff
@@ -222,32 +253,88 @@ def __init__(
...
+        plot_study_run_times
+        plot_study_unit_counts
+        plot_study_performances
+        plot_study_agreement_matrix
+
+    Parameters
+    ----------
+    study: GroundTruthStudy
+        A study object.
+    case_keys: list or None
```
Comment (zm711) [inline src/spikeinterface/widgets/gtstudy.py:308]:
```suggestion
    case_keys: list or None, default: None
```

File: src/spikeinterface/widgets/gtstudy.py [hunk 9]
```diff
@@ -176,43 +173,77 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
...
-    mode: str
-        Which mode in "swarm"
     case_keys: list or None
         A selection of cases to plot, if None, then all.
-
+    ordered: bool
+        Order units with best agreement scores.
+        This enable to see agreement on a diagonal.
+    count_text: bool
+        If True counts are displayed as text
```
Comment (zm711) [inline src/spikeinterface/widgets/gtstudy.py:240]:
```suggestion
```

I don't see `count_text` as a potential parameter.

File: src/spikeinterface/widgets/gtstudy.py [hunk 10]
```diff
@@ -140,16 +134,18 @@ class StudyPerformances(BaseWidget):
     study: GroundTruthStudy
         A study object.
     mode: str
-        Which mode in "swarm"
+        Which mode in "ordered", "snr", "swarm"
+    performance_names: list
```
Comment (zm711) [inline src/spikeinterface/widgets/gtstudy.py:138]:
```suggestion
    performance_names: list or tuple, default: ('accuracy', 'precision', 'recall')
```

**Review by zm711 — COMMENTED** [review]
@samuelgarcia,

sorry one more set of reviews for the docstrings on my side. + one question.

File: src/spikeinterface/widgets/gtstudy.py (hunk 6 above)
Comment (alejoe91) [reply src/spikeinterface/widgets/gtstudy.py:137]: @zm711 I think it's quite common, but we can ref the seaborn page: https://seaborn.pydata.org/generated/seaborn.swarmplot.html

Comment (zm711) [reply src/spikeinterface/widgets/gtstudy.py:137]: Cool thanks. Sounds good!

File: src/spikeinterface/widgets/gtstudy.py [hunk 11]
```diff
@@ -139,17 +133,23 @@ class StudyPerformances(BaseWidget):
...
     study: GroundTruthStudy
         A study object.
-    mode: str
-        Which mode in "swarm"
+    mode: "ordered" | "snr" | "swarm", default: "ordered"
+        Which plot mode to use:
+
+        * "ordered": plot performance metrics vs unit indices ordered by decreasing accuracy (default)
+        * "snr": plot performance metrics vs snr
+        * "swarm": plot performance metrics as a swarm plot (see seaborn.swarmplot for details)
```
Comment (zm711) [inline src/spikeinterface/widgets/gtstudy.py:141]: Thanks That's super helpful @alejoe91

**Review by alejoe91 — APPROVED** [review]
@samuelgarcia we'll need more checks at some point to know if different study steps have run. Also removed some debug plots in the merge.py that were active by default!

**Pull request merged**